### ISSUE TITLE
Problem with reexport items related to a Raw-structure

### ISSUE THREAD

**Issue by mechvel** [agda/agda-stdlib#1534]
This is on lib-1.7,
For example, in the environment of Magma it is natural to use the divisibility relation `_∣_` - as it is usual for algebra,
this is a notion of the Magma "theory".
`_∣_` is defined in Algebra.Definitions.RawMagma,
and the Magma bundle exports rawMagma : RawMagma.
But Magma cannot reexport items from Algebra.Definitions.RawMagma, due to the import loop.
Am I missing something?

Now, to be able to easily get the items (for example,  `_∣_` ) related to Magma, I am forced to introduce
```
module OfMagma  {α α≈} (M : Magma α α≈)
   where
   open Magma M using (... ; setoid; rawMagma)
   open OfSetoid  setoid   public
   open  Algebra.Definitions.RawMagma rawMagma public 
   ...
```
Am I missing something?
A similar complication is for other structures: (Semi)group, (Semiring)Ring, ...
This is a consequence of the whole approach of moving to other modules the item definitions related to a bundle.
I wrote of this several years earlier, argued about this.
Now I do not expect of the team to start refactoring this design.
But may be it worth to add the "Reexport" modules
``module Foo-reexport`` or may be ``module Foo-items`` or ``module Foo-theory`` ?
This is to reexport the items related to a `bundle` Foo, where these items are scattered around in Algebra.Definitions,
Algebra.Properties, (Algebra.Construct ?), and such.
Am I missing something?

**Comment by MatthewDaggitt** [comment]
Yes, so the properties about each individual operation are under `Algebra.Properties.X` as you've discovered. 

> To use other items of Magma, one will be forced to import modules like Algebra.Operations.Foo.

The `Algebra.Operations` folder is entirely deprecated. Nothing in here should be imported.

> May be it worth to add the "Reexport" modules
module Foo-reexport or may be module Foo-items or module Foo-theory ?

Hmm I have no problem with this, but I'm afraid I don't understand your naming convention at all. Could you elaborate on what you mean by `Foo-reexport`? We don't use `-` in names nor lowercase letters.

**Comment by mechvel** [comment]
> The Algebra.Operations folder is entirely deprecated. 

This is nice.

> Could you elaborate on what you mean by Foo-reexport? We don't use - in names nor lowercase letters.

For example: 
a user has ``M : Monoid``. One knows that in a monoid divisibility is reflexive. Also one knows that for a monoid it should  be defined the function `_^_` for powering, because it is easily expressed via `_∙_`.
So the user tries to import these items from somewhere. 
1) they are not exported by the Monoid bundle,
2) one has to guess that  `∣-refl`  is in  Algebra.Properties.Monoid.Divisibility,
3) one has to guess that this  `_^_`  is in  Algebra.Properties.Monoid.Foo?
    - I do not know where does it currently reside, but may be it is somewhere.
   
I suggest to introduce  the module `Algebra.OfMonoid` parameterized by (M : Monoid)
which reexports all the items 
from Monoid, 
from Algebra.Properties.Monoid.Divisibility,
from Algebra.Properties.Monoid.Foo?, 
and so on.
With this, the user would not search among subfolders of the `Algebra` folder. Instead one would write, for example, 
```
open Algebra.OfMonoid M using ( _∙_;  _^_; _∣_ ; ∣-refl)   
```
Because it is known traditionally from theory that these items are related to Monoid, and one does not need to search through various subfolders.
Here `_∣_` has to be reexported from OfMagma, where OfMagma to be introduced earlier similarly as OfMonoid.

Similarly, OfSemigroup and OfGroup are needed, and so on.
This will add the following files to the `Algebra/` folder:
OfMagma.agda, OfSemigroup.agda, OfMonoid.agda, OfGroup.agda, OfSemiring.agds, and so on. 
 
I expect this will make it easier to search for import.
And this is backwards compatible.
?
